src/Microsoft.AspNetCore.Dispatcher/Patterns/InlineRouteParameterParser.cs

```diff
@@ -126,7 +116,7 @@ public static TemplatePart ParseRouteParameter(string routeParameter)
                             case null:
                                 state = ParseState.End;
                                 var constraintText = routeParameter.Substring(startIndex, currentIndex - startIndex);
-                                inlineConstraints.Add(new InlineConstraint(constraintText));
+                                inlineConstraints.Add(ConstraintReference.Create(constraintText));
```

Comment:
What's the benefit of this pattern of .Create vs the ctor/new pattern?

Reply:
There are two reasons:
- Gives you the flexibility to return a subclass in the future if the design changes
- Can come up with more meaningful names (Look at update) when there are multiple ways to create something
- Consistency with one of the other parts of the system that does similar things, but for a better reason